optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,733 @@
+#  Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+import logging
+import os
+import warnings
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Dict, Optional, Union
+
+import numpy as np
+import openvino
+import torch
+from huggingface_hub import hf_hub_download
+from huggingface_hub.constants import HUGGINGFACE_HUB_CACHE
+from torch.nn import functional as F
+from transformers import (
+    AutoConfig,
+    CLIPConfig,
+    PretrainedConfig,
+)
+from transformers.file_utils import add_start_docstrings
+from transformers.modeling_outputs import ModelOutput
+from transformers.models.clip.modeling_clip import CLIPOutput
+
+from optimum.exporters.tasks import TasksManager
+
+from ...exporters.openvino import main_export
+from ..utils.modeling_utils import OpenClipForZeroShotImageClassification
+from .configuration import OVConfig, OVWeightQuantizationConfig
+from .modeling import MODEL_START_DOCSTRING, OVModel
+
+
+logger = logging.getLogger(__name__)
+
+
+class OVModelOpenCLIPBase(OVModel):
+    OPEN_CLIP_CONFIG_NAME = "open_clip_config.json"
+    CONFIG_NAME = "config.json"
+
+    def __init__(self, model=None, config=None, **kwargs):
+        super().__init__(model, config, **kwargs)
+
+    @classmethod
+    def _load_config(
+        cls,
+        config_name_or_path: Union[str, os.PathLike],
+        revision: Optional[str] = None,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        force_download: bool = False,
+        subfolder: str = "",
+        trust_remote_code: bool = False,
+        local_files_only: bool = False,
+    ) -> PretrainedConfig:
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
```

Comment:
shouldn't be needed (already present in from_pretrained)
```suggestion
        token: Optional[Union[bool, str]] = None,
        force_download: bool = False,
        subfolder: str = "",
        trust_remote_code: bool = False,
        local_files_only: bool = False,
    ) -> PretrainedConfig:
```